src/Microsoft.AspNetCore.Razor.Tools/CommandBase.cs

```diff
@@ -20,6 +21,8 @@ protected CommandBase(Application parent, string name)
 
             base.Parent = parent;
             Name = name;
+            Out = output ?? Out;
+            Error = error ?? Error;
```

Comment:
I really don't get how all of this rejiggering of properties vs parameters helps. The properties are still there, and they are still mutable, but in some cases they are ignored?

Reply:
How much output is there going to be?

Reply:
We set the value on the top level `CommadLineApplication` but that doesn't affect any new Commands that are created. This makes them both use the same inputs and outputs

Reply:
> How much output is there going to be?

Output will just be whatever the CLI tool would show (`DiscoverCommand` and `GenerateCommand`). All the other server specific logging like "Request received" etc will not show up here.